server/server.go

```diff
@@ -304,22 +303,15 @@ type txpoolHub struct {
 }
 
 func (t *txpoolHub) GetNonce(root types.Hash, addr types.Address) uint64 {
+	// TODO: Use a function that returns only Account
 	snap, err := t.state.NewSnapshotAt(root)
 	if err != nil {
 		return 0
```

Comment:
shall we log the error?

Reply:
here and below

Reply:
These functions (both for txpoolHub and jsonrpcHub) happen because of user input (send a transaction to the pool, or make a Jsonrpc request). I think that if we log the error, someone could just spam the console of the client.

Reply:
On the other hand, not logging errors with user data makes debugging really hard. we can limit data to log + data logs rotation

Reply:
Let's address this in a follow-up PR. I think this is part of a bigger convo on what to do with errors in general from queries in the JsonRPC-API (I will create a Jira task).

My strategy would be to keep these functions on `server` as they are and have a special `Debug/Trace` on the Json-RPC Api that logs the errors (**only**) if Debug is enabled.

There was a situation before on the `txpool` where a Debug log was accounting for much of the memory usage on the pool because of string manipulation. So, I want to be really careful about these specific debugging scenarios.

Those that sound good to you?

Reply:
works for me. maybe we can, later, have an observability epic or something to check all errors and their's contexts, metrics and tracing.